### ISSUE TITLE
Fix Request for AvoidFutureJoinWithoutTimeout

### ISSUE THREAD

**Issue by jborgers** [jborgers/PMD-jPinpoint-rules#210]
Mixed up with String.join in a stream, example:
```
List<String> pairs = Arrays.stream(topic.split(","))
   .map(item -> item.trim().split("="))
   .filter(itemPair -> "CN".equalsIgnoreCase(itemPair[0]) || "O".equalsIgnoreCase(itemPair[0]))
   .map(itemPair -> String.join(":", itemPair)) // false positive
   .collect(Collectors.toList());
return String.join(" --- ", pairs);

```

**Comment by jborgers** [comment]
duplicate of #198
